sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/system/SystemStatusListener.java

```diff
@@ -54,7 +60,21 @@ public void run() {
              * observed. All values between 0.0 and 1.0 are possible depending of the activities going on in the
              * system. If the system recent cpu usage is not available, the method returns a negative value.
              */
-            currentCpuUsage = osBean.getSystemCpuLoad();
+            double systemCpuUsage = osBean.getSystemCpuLoad();
+
+            // calculate process cpu usage to support application running in container environment
+            RuntimeMXBean runtimeBean = ManagementFactory.getPlatformMXBean(RuntimeMXBean.class);
+            long newProcessCpuTime = osBean.getProcessCpuTime();
+            long newProcessUpTime = runtimeBean.getUptime();
+            int cpuCores = osBean.getAvailableProcessors();
+            long processCpuTimeDiffInMs = TimeUnit.NANOSECONDS
+                    .toMillis(newProcessCpuTime - processCpuTime);
+            long processUpTimeDiffInMs = newProcessUpTime - processUpTime;
+            double processCpuUsage = (double) processCpuTimeDiffInMs / processUpTimeDiffInMs / cpuCores;
+            processCpuTime = newProcessCpuTime;
+            processUpTime = newProcessUpTime;
+
+            currentCpuUsage = Math.max(processCpuUsage, systemCpuUsage);
```

Comment:
And i think if we calculate the cpu usage by ourselves we can abandon `systemCpuUsage` totally.

Reply:
They are both needed. The application performance could be affected if low processCpuUsage but high systemCpuUsage.

Reply:
> They are both needed. The application performance could be affected if low `processCpuUsage` but high `systemCpuUsage`.

In my logic in the scenario you mentioned slow down current application may not make things better and it also can use other dimension like `global avgRT` indicating whether should be limited traffic? After all the rules are written for applications.

eg.  the system cpu load keeps on about 0.8 but process use only 0.1

So what's your opinion?

Reply:
> > They are both needed. The application performance could be affected if low `processCpuUsage` but high `systemCpuUsage`.
> 
> In my logic in the scenario you mentioned slow down current application may not make things better and it also can use other dimension like `global avgRT` indicating whether should be limited traffic? After all the rules are written for applications.
> 
> eg. the system cpu load keeps on about 0.8 but process use only 0.1
> 
> So what's your opinion?

`systemCpuUsage` is appliable for physical machine which might running multiple applications, on the other hand, `processCpuUsage` is for docker environment which running single application. so I use the max value between them to compatiable for both physical and docker environments.

of course, there are many other dimensions, but I think we can keep `systemCpuUsage` and leave the options open to adopters.

Reply:
Okay i don't insist on it and we can keep it for backward compatibility.